gcp_variant_transforms/beam_io/vcf_estimate_io.py

```diff
@@ -0,0 +1,217 @@
+
+      header_size += len(header_line.decode('UTF-8').encode("UTF-8"))
+      # use the last header line to extract samples.
+      calls = header_line.split()[8:] # remove #CHROME..INFO mandatory fields
+      samples = (
+          calls if (not calls or calls[0] != 'FORMAT') else calls[1:])
+
+      first_record = file_to_read.readline()
+      while not first_record or not first_record.strip():
+        first_record = file_to_read.readline()
+
+      size_in_bytes = FileSystems.match(
+          [file_name])[0].metadata_list[0].size_in_bytes
+      all_lines_size = size_in_bytes
+      if not isinstance(file_to_read, CompressedFile):
+        # TODO(tneymanov): Find a better solution to handling compressed files.
+        all_lines_size -= header_size
+      line_size = len(first_record.decode('UTF-8').encode("UTF-8"))
+      estimated_line_count = float(all_lines_size) / line_size
+
+    yield VcfEstimate(file_name=file_name,
+                      samples=samples,
+                      estimated_line_count=estimated_line_count,
+                      size_in_bytes=size_in_bytes)
+
+
+class GetEstimates(PTransform):
+  """A :class:`~apache_beam.transforms.ptransform.PTransform` for reading the
+  vcf files, finding the last header line and first data line and to extract
+  estimates from them.
+  """
+
+  def __init__(
+      self,
+      file_pattern,  # type: str
+      compression_type=CompressionTypes.AUTO,  # type: str
+      validate=True,  # type: bool
+      **kwargs  # type: **str
+      ):
+    # type: (...) -> None
+    """Initialize the :class:`GetEstimates` transform.
+
+    Args:
+      file_pattern: The file path to read from either as a single file or a glob
+        pattern.
+      compression_type: Used to handle compressed input files.
+        Typical value is :attr:`CompressionTypes.AUTO
+        <apache_beam.io.filesystem.CompressionTypes.AUTO>`, in which case the
+        underlying file_path's extension will be used to detect the compression.
+      validate: Flag to verify that the files exist during the pipeline creation
+        time.
+    """
+    super(GetEstimates, self).__init__(**kwargs)
+    self._source = VcfEstimateSource(
+        file_pattern,
+        compression_type,
+        validate=validate)
+
+  def expand(self, pvalue):
+    return pvalue.pipeline | Read(self._source)
+
+
+def _create_vcf_estimate_source(file_pattern=None,
+                                compression_type=None):
+  return VcfEstimateSource(file_pattern=file_pattern,
+                           compression_type=compression_type)
+
+
+class GetAllEstimates(PTransform):
+  """A :class:`~apache_beam.transforms.ptransform.PTransform` for reading the
+  vcf files, finding the last header line and first data line and to extract
+  estimates from them.
+
+  This transform should be used when reading from massive (>70,000) number of
+  files.
+  """
+
+  DEFAULT_DESIRED_BUNDLE_SIZE = 64 * 1024 * 1024  # 64MB
```

Comment:
Where does this value come from?

Reply:
If it's the BEAM default is it exposed at all?  i.e., could we grab it from the underlying reader instead of having it duplicated here?